flake.nix

```diff
@@ -5,27 +5,76 @@
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   };
 
-  outputs = { self, flake-utils, naersk, nixpkgs }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let
+  outputs = {
+    self,
+    flake-utils,
+    naersk,
+    nixpkgs,
+  }:
+    flake-utils.lib.eachDefaultSystem (
+      system: let
         pkgs = (import nixpkgs) {
           inherit system;
         };
+        inherit (pkgs) lib;
+
         naersk' = pkgs.callPackage naersk {};
-        nativeBuildInputs = with pkgs; [ pkg-config ];
-      in rec {
-        packages.default = naersk'.buildPackage {
-          inherit nativeBuildInputs;
-          src = ./.;
-          buildInputs = with pkgs; [ openssl ] ++ (pkgs.lib.lists.optionals (stdenv.isDarwin) [ darwin.apple_sdk.frameworks.SystemConfiguration ]);
+        nativeBuildInputs = with pkgs; [pkg-config];
+
+        darwin_frameworks = with pkgs.darwin.apple_sdk.frameworks; [
+          Security
+          SystemConfiguration
+        ];
+
+        dev_tools = with pkgs; [
+          cargo
+          rustc
+        ];
+
+        dependencies = with pkgs;
+          [
+            libgit2
+            openssl
+          ]
+          ++ lib.lists.optionals stdenv.isDarwin darwin_frameworks;
+
+        env_vars = {
+          LIBGIT2_NO_VENDOR = 1;
+          OPENSSL_NO_VENDOR = 1;
         };
-        devShells.default = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; nativeBuildInputs ++ [
-            cargo
-            rustc
-          ];
+      in rec {
+        # NB: if this does not build and you need to modify the file,
+        #     please ensure you also make the corresponding changes in the devshell
+        packages.default = naersk'.buildPackage ({
```

Comment:
any chance we could take this opportunity to build with Crane and not Naersk? I've seen a bunch of internal issues which I _think_ are triggered by building with Naersk and not Crane.

Reply:
[Using this as a starting point](https://github.com/ipetkov/crane/blob/master/examples/quick-start/flake.nix)